Build failure on 32-bit

See e.g. https://github.com/r-universe/r-spatial/runs/2762608125?check_suite_focus=true

Looks like a type error in the line with `array->Read()` for 32-bit platforms.


```
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: mdim.o:mdim.cpp:(.text+0x4dc): undefined reference to `GDALAbstractMDArray::Read(unsigned long long const*, unsigned int const*, long long const*, long long const*, GDALExtendedDataType const&, void*, void const*, unsigned int) const'
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: mdim.o:mdim.cpp:(.text+0x9bf): undefined reference to `GDALAbstractMDArray::Read(unsigned long long const*, unsigned int const*, long long const*, long long const*, GDALExtendedDataType const&, void*, void const*, unsigned int) const'
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: mdim.o:mdim.cpp:(.text+0x2810): undefined reference to `GDALAbstractMDArray::Read(unsigned long long const*, unsigned int const*, long long const*, long long const*, GDALExtendedDataType const&, void*, void const*, unsigned int) const'
collect2.exe: error: ld returned 1 exit status
```

--- a/src/mdim.cpp
+++ b/src/mdim.cpp
@@ -9,7 +9,11 @@
 
 using namespace Rcpp;
 
-#if GDAL_VERSION_NUM >= 3010000
+#if defined(__MINGW32__) && !defined(__MINGW64__)
+#define WIN32BIT
+#endif
+
+#if GDAL_VERSION_NUM >= 3010000 && !(defined(WIN32BIT))
 List get_dimension_values(std::shared_ptr<GDALMDArray> array) {
 	size_t nValues = 1;
 	std::vector<size_t> anCount;
@@ -230,9 +234,9 @@ List write_mdim(List x, CharacterVector file, List dimensions, CharacterVector u
 
 #else
 List read_mdim(CharacterVector file, CharacterVector array_names, CharacterVector oo) {
-	stop("requires GDAL >= 3.1.0");
+	stop("requires GDAL >= 3.1.0 and 64-bit");
 }
 List write_mdim(List x, CharacterVector file, List dimensions, CharacterVector units) {
-	stop("requires GDAL >= 3.1.0");
+	stop("requires GDAL >= 3.1.0 and 64-bit");
 }
 #endif

